### PR TITLE
Port _guess_toolchain_kwargs_from_python_config away from distutils

### PR DESCRIPTION
**Pull request by alexfikl**
At least on my system, all these variables seemed to return the same things, so hopefully it doesn't break too much. I also ran the `loopy` tests and they still pass.

### CODE REVIEW

File: codepy/toolchain.py [hunk 1]
```diff
@@ -368,23 +368,16 @@ class ToolchainGuessError(Exception):
...
-    make_vars = parse_makefile(get_makefile_filename())
-
-    cc_cmdline = (make_vars["CXX"].split()
-            + make_vars["CFLAGS"].split()
-            + make_vars["CFLAGSFORSHARED"].split())
-    object_names = [
-            oname for oname in make_vars["OBJECT_OBJS"].split()
-            if "(" not in oname and ")" not in oname]
-
-    object_suffix = ".{}".format(object_names[0].split(".")[1])
```
Comment (alexfikl) [inline codepy/toolchain.py:387]: This just tries to get the extension for objects on the system, right?

It seems `sysconfig` has a dumber `parse_makefile` that doesn't parse multiline variables. This switched to check `MODOBJS`, which seems to be the only variable in the makefile that's defined on a single line :\

File: codepy/toolchain.py [hunk 2]
```diff
@@ -400,26 +393,23 @@ def strip_prefix(pfx, value):
...
-    for lib in make_vars["LIBS"].split():
+    for lib in config_vars["LIBS"].split():
         if lib.startswith("-l"):
-            libraries.append(strip_prefix("-l", lib))
+            libraries.append(lib[2:])
         else:
             cflags.append(lib)
 
+    ld, *ldflags = config_vars["LDSHARED"].split()
+    so_ext = os.path.splitext(config_vars["EXT_SUFFIX"])[-1]
```
Comment (alexfikl) [inline codepy/toolchain.py:403]: The previous version seemed to assume that `make_vars["SO"]` returned an extension, but at least on my system it doesn't exist at all. `config_vars["SO"]` exists though (from `sysconfig`), but that returns the `cpython-[arch].so`.

It also says that "SO" is deprecated in favor of "EXT_SUFFIX".

Comment (alexfikl) [reply codepy/toolchain.py:403]: Actually, it's `.cpython-[arch].so`.. was that the intended full suffix here?

Comment (inducer) [reply codepy/toolchain.py]: I think that's OK. As long as CPython looks for `xyz.cpython-[arch].so` for `import xyz`, that's all that's needed.

**Comment by inducer** [comment]
My biggest worry here is Mac compatibility. Could you add a CI that tries that? (Maybe on `main` first, to see what passes there?)

**Pull request merged**